hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/impl/SkipOnConcurrentExecutionDecorator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.executionservice.impl;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Decorator to prevent concurrent task execution.
+ *
+ * If this task is running on a thread and another thread calls attempt the execute it concurrently
+ * then the 2nd execution will be skipped.
+ *
+ */
+public class SkipOnConcurrentExecutionDecorator implements Runnable {
+    private final AtomicBoolean isAlreadyRunning = new AtomicBoolean();
```

Comment:
`AtomicIntegerFieldUpdater` tends to be a tiny bit slower in current HotSpot. (I think the reason is that HotSpot does not fully trust final fields inside `Atomic*FieldUpdater`) 

I would have used the `Atomic*FieldUpdater` if the tasks were created with high rate. However here I assume a component schedules a task once only - during initialization/start. Hence the amount of objects created is not growing over time. 

If you still think it's good to use the Update I can change it, but I see little reasons to do so.